torch/csrc/Size.cpp

```diff
@@ -23,10 +23,19 @@ static PyObject * THPSize_pynew(PyTypeObject *type, PyObject *args, PyObject *kw
   if (self) {
     for (Py_ssize_t i = 0; i < PyTuple_Size(self); ++i) {
       PyObject *item = PyTuple_GET_ITEM(self.get(), i);
-      if (!THPUtils_checkLong(item)) {
-        return PyErr_Format(PyExc_TypeError, "torch.Size() takes an iterable of 'int' (item %zd is '%s')",
-            i, Py_TYPE(item)->tp_name);
+      if (THPUtils_checkLong(item)) {
```

Comment:
I've since added checkIndex / unpackIndex to THPUtils, I think you want to use those instead, but you should double check.